GOT-StudentVersion/GOT-StudentVersion/Controllers/GOTViewController.swift

```diff
@@ -0,0 +1,211 @@
+        
+        switch segueIdentifer {
+        case "segueToDetail":
+            guard let destVC = segue.destination as? detailViewController else {
+                fatalError("Unexpected segue VC")
+            }
+            guard let selectedIndexPath = tableView.indexPathForSelectedRow else {fatalError("No row selected")
+            }
+            let currentEpisode = EpisodeSearchResults[selectedIndexPath.section][selectedIndexPath.row]
+            destVC.myCurrentGOTEpisode = currentEpisode
+        default:
+            fatalError("unexpected segue identifies")
+        }
+    }
+    
+    //MARK: -- View methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTableView()
+        
+    }
+    
+    private func configureTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        searchBarOutlet.delegate = self
+        tableView.sectionIndexColor = .white
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        // Hide the navigation bar for current view controller
+        self.navigationController?.isNavigationBarHidden = true;
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        // Show the navigation bar on other view controllers
+        self.navigationController?.isNavigationBarHidden = false;
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+
+//MARK: -- Table Datasource Methods
+extension GOTViewController: UITableViewDataSource {
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return EpisodeSearchResults.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section{
+        case 0: return "Season 1"
+        case 1: return "Season 2"
+        case 2: return "Season 3"
+        case 3: return "Season 4"
+        case 4: return "Season 5"
+        case 5: return "Season 6"
+        case 6: return "Season 7"
+        case 7: return "Season 8"
+        default: return ""
+        }
+    }
+    
+    func sectionIndexTitles(for tableView: UITableView) -> [String]? {
+        if (searchBarOutlet.isFirstResponder) {
+            return nil
+        } else {
+            return ["1", "2", "3", "4", "5", "6", "7", "8"]
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return EpisodeSearchResults[section].count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let currentEpisode: GOTEpisode = EpisodeSearchResults[indexPath.section][indexPath.row]
+        
+        switch indexPath.section % 2 {
```

Comment:
Instead of doing a switch with a default cause just use an if else. This way you wouldn't have to return an empty UITableViewCell